C/c4Document.cc

```diff
@@ -41,9 +41,18 @@ C4Document::C4Document(C4Collection *collection, alloc_slice docID_)
 :_collection(asInternal(collection))
 ,_docID(move(docID_))
 {
+
+// GCC will (probably falsely) flag the below as a null directive
```

Comment:
Oh! GCC noticed that `docID_` was moved from (on line 42) so it's known to be null.
The real fix is to change `docID_` to `_docID` on line 52; that should take care of the warning too.